src/main/java/core/basesyntax/strategy/OperationStrategy.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.TransactionHandler;
+import core.basesyntax.service.impl.BalanceTransactionHandler;
+import core.basesyntax.service.impl.PurchaseTransactionHandler;
+import core.basesyntax.service.impl.ReturnTransactionHandler;
+import core.basesyntax.service.impl.SupplyTransactionHandler;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationStrategy {
+    private Map<String, TransactionHandler> operationMap;
+
+    public OperationStrategy() {
+        operationMap = new HashMap<>();
+        operationMap.put(Operation.BALANCE.getCode(), new BalanceTransactionHandler());
+        operationMap.put(Operation.SUPPLY.getCode(), new SupplyTransactionHandler());
+        operationMap.put(Operation.PURCHASE.getCode(), new PurchaseTransactionHandler());
+        operationMap.put(Operation.RETURN.getCode(), new ReturnTransactionHandler());
```

Comment:
Please move map initialization to Main class and pass ready to use map via constructor